misk-aws/src/main/kotlin/misk/jobqueue/sqs/SqsJobConsumer.kt

```diff
@@ -231,6 +253,71 @@ internal class SqsJobConsumer @Inject internal constructor(
         )
       }
     }
+
+  }
+
+  internal inner class BatchQueueReceiver(
+    queueName: QueueName,
+    private val handler: BatchJobHandler
+  ) : QueueReceiver(queueName) {
+
+    override fun receive(): List<Status> {
+      // Repeatedly long poll for messages until we have batchSize messages.
+      // If there are no messages available, short circuit further requests.
+      val cutoff = TimeSource.Monotonic.markNow() + 20.seconds // TODO make configurable
+      val batch: MutableList<SqsJob> = mutableListOf()
+      while (batch.size < batchSize() && !cutoff.hasPassedNow()) {
+        val messages = fetchMessages(
+          sqsBatchSize = min(10, batchSize() - batch.size),
+          waitTimeSeconds = 5
```

Comment:
I need to change this to `min(secondsPerBatch, secondsUntilCutoff)`. That achieves two things:
* We need to limit `secondsPerBatch` such that we issue at least 5 requests for batchSize 50, 4 requests for batchSize 40, etc. before we reach the cutoff.
* We need `secondsUntilCutoff` to avoid exceeding the max total wait time. It also has the benefit of enabling short polling by setting max total wait time to 0s.